bootstrap/internal/controller/infraenv_controller.go

```diff
@@ -77,10 +101,34 @@ func (r *InfraEnvReconciler) attachISOToAgentBootstrapConfigs(ctx context.Contex
 		}
 
 		// Add ISO to agentBootstrapConfig status
-		agentBootstrapConfig.Status.ISODownloadURL = infraEnv.Status.ISODownloadURL
+		agentBootstrapConfig.Status.ISODownloadURL = downloadURL
 		if err := r.Client.Status().Update(ctx, &agentBootstrapConfig); err != nil {
 			return errors.Wrap(err, "failed to update agentbootstrapconfig")
 		}
 	}
 	return nil
 }
+
+func (r *InfraEnvReconciler) getISOURL(ctx context.Context, originalURL string) (string, error) {
+	if !r.Config.UseInternalImageURL {
+		return originalURL, nil
+	}
+
+	svc := &corev1.Service{}
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: r.Config.ImageServiceName, Namespace: r.Config.ImageServiceNamespace}, svc); err != nil {
+		return "", errors.Wrap(err, "failed to find assisted image service service")
+	}
+
+	if svc.Spec.ClusterIP == "" || len(svc.Spec.Ports) < 1 {
+		return "", fmt.Errorf("failed to get internal image service URL, either cluster IP or Ports were missing from Service")
+	}
+
+	isoURL, err := url.Parse(originalURL)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to parse InfraEnv ISO download URL %s", originalURL)
+	}
+
+	isoURL.Scheme = "http"
+	isoURL.Host = fmt.Sprintf("%s:%s", svc.Spec.ClusterIP, strconv.Itoa(int(svc.Spec.Ports[0].Port)))
```

Comment:
You can probably use `%d` here instead of strconv and a cast.
```suggestion
	isoURL.Host = fmt.Sprintf("%s:%d", svc.Spec.ClusterIP, svc.Spec.Ports[0].Port)
```

Reply:
Ah good point 🤦‍♀️ thank you!! Modified